ansible/roles/test/files/ptftests/arista.py

```diff
@@ -248,24 +266,28 @@ def parse_logs(self, data):
         # first state is Idle, last state is Established
         for events in result_bgp.values():
             if len(events) > 1:
-                assert(events[0][1] != 'Established')
+                first_state = events[0][1]
+                assert first_state != 'Established', 'First BGP state should not be Established, it was {}'.format(first_state)
 
-            assert(events[-1][1] == 'Established')
+            last_state = events[-1][1]
+            assert last_state == 'Established', 'Last BGP state is not Established, it was {}'.format(last_state)
 
-        # verify BGP establishment time between v4 and v6 peer is not more than 20s
+        # verify BGP establishment time between v4 and v6 peer is not more than 40s
         if self.reboot_type == 'warm-reboot':
             estab_time = 0
             for ip in result_bgp:
                 if estab_time > 0:
                     diff = abs(result_bgp[ip][-1][0] - estab_time)
-                    assert(diff <= 20)
+                    assert diff <= 40, 'BGP establishement time between v4 and v6 peer is longer than 40 sec, it was {}'.format(diff)
```

Comment:
1. It may be better to make this BGP establishment time an argument from the advanced-reboot.
And also, advanced-reboot should also get this as an argument from the test that calls it (with a default of 40).

2. If agreed, can we also make 20s default for sonic-quagga, and 40s default for FRR? Any BgP routing stack can be installed on SONiC, and I feel test that test should adjust this accordingly. Thoughts?

Reply:
1. I added a test parameter to control this time with a default of 40 sec
2. I think that quagga is not supported on 201911, 202012 and master. And for 201811 (with quagga) we have a different sonic-mgmt branch. But this needs a clarification from MSFT and also clarification about this requirement on timing.